### PR TITLE
image-sets-view: fixes image-sets-view count sql.

### PR DESCRIPTION
**Pull request by ldjebran**
align the count sql with the one for data.

## Type of change

What is it?

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [ ] Refactor

# Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I run `make pre-commit` to check fmt/vet/lint/test-no-fdo

### CODE REVIEW

File: pkg/services/imagesets.go [hunk 1]
```diff
@@ -102,10 +102,16 @@ func (s *ImageSetsService) GetImageSetsViewCount(tx *gorm.DB) (int64, error) {
...
+	// create a sub query of the latest images and their corresponding image sets
+	latestImagesSubQuery := db.Org(orgID, "").Model(&models.Image{}).Select("image_set_id", "max(id) as image_id").Group("image_set_id")
+	if result := db.OrgDB(orgID, tx, "images").Table("(?) as latest_images", latestImagesSubQuery).
+		Joins("JOIN images on images.id = latest_images.image_id").
+		Joins("JOIN image_sets on image_sets.id = latest_images.image_set_id").
+		Where("image_sets.deleted_at IS NULL").
+		Count(&count); result.Error != nil {
+		log.WithFields(log.Fields{"error": result.Error.Error(), "OrgID": orgID}).Error(
+			"error when getting image sets view data",
+		)
```
Comment (de1987) [inline pkg/services/imagesets.go:114]: Would be great if we have a test to cover this change.
WDYT?

**Review by tomershinhar — DISMISSED** [review]
LGTM

File: pkg/services/imagesets.go (hunk 1 above)
Comment (ldjebran) [reply pkg/services/imagesets.go:114]: Ok will create a test , note that we are only aligning with the data one so nothing new done here.

Comment (ldjebran) [reply pkg/services/imagesets.go:114]: there was one unit-tests for count added 3 case for status , that reflect the functionality.

File: pkg/services/imagesets_test.go [hunk 2]
```diff
@@ -90,6 +92,33 @@ var _ = Describe("ImageSets Service Test", func() {
...
+			Expect(count).To(Equal(int64(1)))
+		})
+
+		It("should return The right image view count  when filtering by status building", func() {
+
+			dbFilter := db.DB.Where("image_sets.name LIKE ? AND images.status =  ?", CommonName+"%", models.ImageStatusBuilding)
+
+			count, err := service.GetImageSetsViewCount(dbFilter)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(count).To(Equal(int64(0)))
```
Comment (tomershinhar) [inline pkg/services/imagesets_test.go:119]: shouldn't count be 1 here? as otherImage0 is with status BUILDING
i might be missing something though

Comment (ldjebran) [reply pkg/services/imagesets_test.go:119]: yes it's not the last one :)

Comment (ldjebran) [reply pkg/services/imagesets_test.go:119]: because otherImage0 is not the last one in image-set , it's not and should not be affected by the filler.

**Review by tomershinhar — APPROVED** [review]
LGTM

**Review by mgold1234 — APPROVED** [review]
/lgtm

File: pkg/services/imagesets.go (hunk 1 above)
Comment (ldjebran) [reply pkg/services/imagesets.go:114]: resolve this as implemented the unittests

**Comment by ldjebran** [comment]
remove do not merge as seems everything ok now , with automation

**Pull request merged**